### ISSUE TITLE
Support deptrac 

### ISSUE THREAD

**Issue by archfz** [EdgedesignCZ/phpqa#229]
It would be nice if [deptrac](https://github.com/qossmic/deptrac) was also support among the tools. It is useful to allow validating against architecturally disallowed dependencies.

**Comment by zdenekdrahos** [comment]
What configuration do you need?

```bash
$ vendor/bin/deptrac analyze --help

Arguments:
  depfile                                          Path to the depfile

Options:
      --formatter=FORMATTER                        Format in which to print the result of the analysis. Possible: ["console", "github-actions", "graphviz", "junit", "table", "xml", "baseline"] (multiple values allowed)
      --no-progress                                Do not show progress bar
      --fail-on-uncovered                          Fails if any uncovered dependency is found
      --report-uncovered                           Report uncovered dependencies
      --report-skipped                             Report skipped violations
      --graphviz-display[=GRAPHVIZ-DISPLAY]        Should try to open graphviz image. [default: true]
      --graphviz-dump-image[=GRAPHVIZ-DUMP-IMAGE]  Path to a dumped png file.
      --graphviz-dump-dot[=GRAPHVIZ-DUMP-DOT]      Path to a dumped dot file.
      --graphviz-dump-html[=GRAPHVIZ-DUMP-HTML]    Path to a dumped html file.
      --junit-dump-xml[=JUNIT-DUMP-XML]            Path to a dumped xml file. [default: "./junit-report.xml"]
      --xml-dump[=XML-DUMP]                        path to a dumped xml file [default: "./deptrac-report.xml"]
      --baseline-dump[=BASELINE-DUMP]              path to a dumped baseline file [default: "./depfile.baseline.yml"]
      --no-cache                                   Disable caching mechanisms
      --cache-file=CACHE-FILE                       [default: ".deptrac.cache"]
```

---

You can try initial version with [console formatter](https://github.com/EdgedesignCZ/phpqa/runs/2573983870?check_suite_focus=true#step:7:131):

```bash
$ composer update edgedesign/phpqa:dev-deptrac
$ phpqa --tools deptrac -v
```

![github actions](https://user-images.githubusercontent.com/7994022/118102362-fcc4dd00-b3d8-11eb-9d36-0d06ea7ad253.png)

**Comment by archfz** [comment]
Well at first its enough if it is able to load the depfile.yaml from root of project

**Comment by zdenekdrahos** [comment]
Yes, root file is used if `deptrac.depfile` is not configured:

```
$ phpqa --tools deptrac:0 -v
 [Edge\QA\Task\ParallelExec] Output for  "/phpqa/vendor/bin/deptrac" analyze  --formatter=console --report-uncovered
  In CouldNotReadFileException.php line 10:
  File "/phpqa/depfile.yaml" cannot be read.

[phpqa]
+---------+----------------+--------------+--------+-------------------+
| Tool    | Allowed Errors | Errors count | Is OK? | HTML report       |
+---------+----------------+--------------+--------+-------------------+
| deptrac | 0              | 1            | x      |                   |
+---------+----------------+--------------+--------+-------------------+

[phpqa] Failed tools: deptrac
```

**Comment by archfz** [comment]
Ok. Will test next week.

**Comment by archfz** [comment]
Tested it's working. Notify when released. As a suggestion I would keep issues open until they are added in an actual release.

**Comment by zdenekdrahos** [comment]
Released in [v1.25](https://github.com/EdgedesignCZ/phpqa/releases/tag/v1.25.0) _([demo](https://edgedesigncz.github.io/phpqa/report/phpqa.html#deptrac))_
